File: python/13.roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:

        mapping = {"M": 1000, "D": 500, "C": 100, "L": 50, "X": 10, "V": 5, "I": 1}

        i = n = len(s) - 1
        result = 0

        while i >= 0:
            if i < n and mapping[s[i]] < mapping[s[i + 1]]:
                result -= mapping[s[i]]
            else:
                result += mapping[s[i]]
            i -= 1
        return result

File: cpp/13.roman_to_integer.cpp
#include <string>
#include <unordered_map>

class Solution {
 public:
    int romanToInt(std::string s) {
        std::unordered_map<char, int> mappings = {
            {'I', 1},   {'V', 5},   {'X', 10},  {'L', 50},
            {'C', 100}, {'D', 500}, {'M', 1000}};

        int size = s.size();
        int i = size - 1;
        int result = 0;

        while (i >= 0) {
            if (i < size - 1 && mappings[s[i]] < mappings[s[i + 1]]) {
                result -= mappings[s[i]];
            } else {
                result += mappings[s[i]];
            }
            --i;
        }
        return result;
    }
};
